kore/src/Kore/Builtin/AssocComm/AssocComm.hs

```diff
@@ -105,7 +99,7 @@ asTermLike
     addTerms terms term = List.foldr concat' term terms
 
     concat' :: TermLike variable -> TermLike variable -> TermLike variable
-    concat' map1 map2 = mkApplySymbol concatSymbol [map1, map2]
+    concat' map1 map2 = mkApplySymbol (fromConcat concatSym) [map1, map2]
```

Comment:
Can `asTermLike` be refactored to avoid using partial functions here and at line 89?

Reply:
see discussion above